Scripts/Python/ki/xKIChat.py

```diff
@@ -401,6 +404,21 @@ def SendMessage(self, message):
     ## Adds a line to the RT chat.
     def AddChatLine(self, player, message, cFlags, forceKI=True):
 
+        playerName = PtGetClientName()
+        playerNameMentionRegex = None
+
+        # Compose the regex differently depending on if player name starts and ends with a word or with special characters
+        if _STARTS_WITH_WORD_REGEX.search(playerName) != None:
+            if _ENDS_WITH_WORD_REGEX.search(playerName) != None:
+                playerNameMentionRegex = re.compile(fr"\b({re.escape(playerName)})\b", re.IGNORECASE)
+            else:
+                playerNameMentionRegex = re.compile(fr"\b({re.escape(playerName)})(?=$|\W)", re.IGNORECASE)
+        else:
+            if _ENDS_WITH_WORD_REGEX.search(playerName) != None:
+                playerNameMentionRegex = re.compile(fr"(?=^|\W)({re.escape(playerName)})\b", re.IGNORECASE)
+            else:
+                playerNameMentionRegex = re.compile(fr"(?=^|\W)({re.escape(playerName)})(?=$|\W)", re.IGNORECASE)
```

Comment:
@Deledrius You're a bit better with regexes than I am. Any thoughts about this?

Reply:
I've posted an alternative at ZarothYe/Plasma#1 for review.

Reply:
The new method seems more robust to me. :)